File: csharp-project/RekindleBookStoreAqa/RekindleTaf/Model/Factory/BookstoreFactory.cs
using System.Collections.Generic;
using Bogus;
using RekindleTaf.Exceptions;
using RekindleTaf.Utils;

namespace RekindleTaf.Model.Factory;

public class BookstoreFactory : IFactory<Bookstore>
{
    private static readonly Faker<Bookstore> BookstoreActiveFaker = new Faker<Bookstore>()
        .RuleFor(bookstore => bookstore.IsActive, () => true)
        .RuleFor(bookstore => bookstore.Name, (faker) => faker.Company.CompanyName());
    
    private static readonly Faker<Bookstore> BookstoreInActiveFaker = new Faker<Bookstore>()
        .RuleFor(bookstore => bookstore.IsActive, () => false)
        .RuleFor(bookstore => bookstore.Name, (faker) => faker.Company.CompanyName());
    
    private IDictionary<IValueKey, Supplier<Bookstore>>
        Map = new Dictionary<IValueKey, Supplier<Bookstore>>()
        {
            {BookstoreKey.StdBookstoreActive, () => BookstoreActiveFaker.Generate()},
            {BookstoreKey.StdBookstoreInactive, () => BookstoreInActiveFaker.Generate()}
        };

    public Bookstore Create(IValueKey key)
    {
        if (Map.TryGetValue(key, out Supplier<Bookstore>? value))
        {
            return value.Invoke();
        }
        throw new ValueObjectNotFoundException($"Bookstore identified by {key} has not yet been implemented");
    }
}

File: typescript-project/rekindle-book-store-aqa/src/model/bookstore-factory.ts
import {Bookstore} from "@/model/domain-model";
import { faker } from '@faker-js/faker';
import {Factory} from "@/model/factory";

export const BookstoreKey = {
    stdBookstoreActive: Symbol('StdBookstoreActive'),
    stdBookstoreInactive: Symbol('StdBookstoreInactive')
};

export class BookstoreFactory implements Factory<Bookstore> {

    private readonly map: Map<Symbol, Supplier<Bookstore>> = new Map([
        [BookstoreKey.stdBookstoreActive, () => this.createActiveBookstore()],
        [BookstoreKey.stdBookstoreInactive, () => this.createInActiveBookstore()]
    ]);

    private createActiveBookstore() : Bookstore {
        return {
            name: faker.company.name(),
            isActive: true
        }
    }

    private createInActiveBookstore() : Bookstore {
        return {
            name: faker.company.name(),
            isActive: false
        }
    }

    create(key: Symbol): Bookstore {
        const supplier = this.map.get(key);
        if (supplier) {
            return supplier();
        }
        throw new Error(`Bookstore identified by ${key.toString()} has not yet been implemented`);
    }

}